framework/tlvf/src/src/WSC/AttrList.cpp

```diff
@@ -0,0 +1,344 @@
+            }
+            break;
+        case ATTR_OS_VERSION:
+            if (!addAttr<cWscAttrOsVersion>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrOsVersion";
+                return false;
+            }
+            break;
+        case ATTR_PUBLIC_KEY:
+            if (!addAttr<cWscAttrPublicKey>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrPublicKey";
+                return false;
+            }
+            break;
+        case ATTR_RF_BANDS:
+            if (!addAttr<cWscAttrRfBands>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrRfBands";
+                return false;
+            }
+            break;
+        case ATTR_SERIAL_NUMBER:
+            if (!addAttr<cWscAttrSerialNumber>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrSerialNumber";
+                return false;
+            }
+            break;
+        case ATTR_WSC_STATE:
+            if (!addAttr<cWscAttrWscState>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrWscState";
+                return false;
+            }
+            break;
+        case ATTR_UUID_E:
+            if (!addAttr<cWscAttrUuidE>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrUuidE";
+                return false;
+            }
+            break;
+        case ATTR_VENDOR_EXTENSION:
+            if (!addAttr<cWscVendorExtWfa>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscVendorExtWfa";
+                return false;
+            }
+            break;
+        case ATTR_VERSION:
+            if (!addAttr<cWscAttrVersion>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrVersion";
+                return false;
+            }
+            break;
+        case ATTR_PRIMARY_DEV_TYPE:
+            if (!addAttr<cWscAttrPrimaryDeviceType>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrPrimaryDeviceType";
+                return false;
+            }
+            break;
+        case ATTR_REGISTRAR_NONCE:
+            if (!addAttr<cWscAttrRegistrarNonce>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrRegistrarNonce";
+                return false;
+            }
+            break;
+        case ATTR_UUID_R:
+            if (!addAttr<cWscAttrUuidR>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrUuidR";
+                return false;
+            }
+            break;
+        case ATTR_ENCR_SETTINGS:
+            if (!addAttr<cWscAttrEncryptedSettings>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrEncryptedSettings";
+                return false;
+            }
+            break;
+        case ATTR_AUTHENTICATOR:
+            if (!addAttr<cWscAttrAuthenticator>()) {
+                TLVF_LOG(ERROR) << "Failed to add cWscAttrAuthenticator";
+                return false;
+            }
+            break;
+        // TODO these are not expected attributes (part of config data)
```

Comment:
Although I commented on the missing FALLTHROUGH attribute, maybe it's better to simply remove unexpected cases.
All unexpected attributes will be caught by the "default" case anyway.